### ISSUE TITLE
msign doesn't work on paths with # in them

### ISSUE THREAD

**Issue by davepacheco** [joyent/node-manta#219]
If you create an object with a '#' in it and try to msign a URL to it, it omits everything after the '#'.  I think it's because we use the "pathname" field of Node's url.parse():

```
{ protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: '#bar',
  search: null,
  query: null,
  pathname: '/dap/stor/foo',
  path: '/dap/stor/foo',
  href: '/dap/stor/foo#bar' }
```

As we can see, though, that field leaves off everything after the "#".  "href" isn't right either in the case that the input has a question mark.  @jclulow points out that we probably don't want to use url.parse at all, but rather just encode the argument given.

**Comment by davepacheco** [comment]
To be clear, "mget" and "mput" work fine on paths with a '#' and space in them (for examples).

**Comment by jclulow** [comment]
So, we should not be parsing the input as a URL at all.  The manual page states that the command accepts `OBJECT...`, a _"list of objects (or directories)"_.  The usage information in the command suggests it accepts `path...`.  Neither of these mention URLs, and indeed we do nothing with an _actual_ URL except (apparently incorrectly) clip the path out of it.

I will rip out the URL parsing step, which will cause us to fail unless you supply an actual Manta path.

**Comment by jclulow** [comment]
Fixed in 12f4c84, released as manta@1.5.0.
